src/System.CommandLine.Subsystems/Validation/Validator.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.CommandLine.Parsing;
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.CommandLine.Validation;
+
+public abstract class Validator
+{
+    public Validator(string name, Type valueConditionType, params Type[] moreValueConditionTypes)
```

Comment:
This should be internal since we only expect CommandValidator and ValueValidator to subclass this

Reply:
CommandValidator and ValueValiditor are base classes for other Validators. We can discuss whether having a Validation and a ValueCondition extensibility point, but if we leave Validators, I think we need to leave it public.